src/document.jl

```diff
@@ -1,4 +1,89 @@
 type Document
-    data::Vector{UInt8}
+    _content::String
+    _line_offsets::Nullable{Vector{Int}}
     blocks::Vector{Any}
+
+    function Document(text::AbstractString)
+        return new(text, Nullable{Vector{Int}}(), [])
+    end
+end
+
+function get_text(doc::Document)
+    return doc._content
+end
+
+function get_line(doc::Document, line::Int)
+    line_offsets = get_line_offsets(doc)
+
+    if length(line_offsets)>0
+        start_offset = line_offsets[line]
+        if length(line_offsets)>line
+            end_offset = line_offsets[line+1]-1
+        else
+            end_offset = endof(doc._content)
+        end
+        return doc._content[start_offset:end_offset]
+    else
+        return ""
+    end
+end
+
+function get_offset(doc::Document, line::Int, character::Int)
+    line_offsets = get_line_offsets(doc)
+    current_offset = line_offsets[line]
```

Comment:
I'm hitting an out of bounds error following:
1. new document
2. new line
3. any character